src/main/java/uk/gov/hmcts/reform/migration/service/RespondentsDataMigrationService.java

```diff
@@ -0,0 +1,149 @@
+
+        /// ADDITIONAL RESPONDENT
+
+        List<Map<String, Object>> additionalRespondents =
+            (List<Map<String, Object>>) objectMapper.convertValue(defaultIfNull(respondents.get("additional"), new ArrayList<>()), List.class);
+
+        List<Respondent> migratedRespondentCollection = additionalRespondents.stream()
+            .map(respondent ->
+                migrateIndividualRespondent(objectMapper.convertValue(respondent.get("value"), OldRespondent.class)))
+            .collect(toList());
+
+        // ADD FIRST RESPONDENT TO ADDITIONAL RESPONDENT LIST
+
+        migratedRespondentCollection.add(migratedFirstRespondent);
+
+        List<Map<String, Object>> newStructure = new ArrayList<>();
+
+        /// BUILD NEW STRUCTURE
+
+        migratedRespondentCollection.forEach(item -> {
+            Map map = objectMapper.convertValue(item, Map.class);
+
+            newStructure.add(ImmutableMap.of(
+                "id", UUID.randomUUID().toString(),
+                "value", map));
+        });
+
+        log.info("returning new structure {}", newStructure);
+
+        return newStructure;
+    }
+
+    private Respondent migrateIndividualRespondent(OldRespondent or) {
+        log.info("migrating respondent {}", or);
+
+        Address.AddressBuilder addressBuilder = Address.builder();
+        if (!isEmpty(or.getAddress())) {
+            addressBuilder.addressLine1(defaultIfBlank(or.getAddress().getAddressLine1(), null));
+            addressBuilder.addressLine2(defaultIfBlank(or.getAddress().getAddressLine2(), null));
+            addressBuilder.addressLine3(defaultIfBlank(or.getAddress().getAddressLine3(), null));
+            addressBuilder.postTown(defaultIfBlank(or.getAddress().getPostTown(), null));
+            addressBuilder.postcode(defaultIfBlank(or.getAddress().getPostcode(), null));
+            addressBuilder.county(defaultIfBlank(or.getAddress().getCounty(), null));
+            addressBuilder.country(defaultIfBlank(or.getAddress().getCountry(), null));
+        }
+        Address address = addressBuilder.build();
+
+        TelephoneNumber.TelephoneNumberBuilder telephoneNumberBuilder = TelephoneNumber.builder();
+        telephoneNumberBuilder.telephoneNumber(defaultIfBlank(or.getTelephone(), null));
+        TelephoneNumber telephoneNumber = telephoneNumberBuilder.build();
+
+        RespondentParty.RespondentPartyBuilder partyBuilder = RespondentParty.builder();
+        partyBuilder.partyID(UUID.randomUUID().toString());
+
+        if (!isEmpty(or.getName())) {
+            partyBuilder.firstName(splitName(or.getName()).get(0));
+
+            if (splitName(or.getName()).size() > 1) {
+                partyBuilder.lastName(splitName(or.getName()).get(1));
+            }
+        }
+
+        partyBuilder.dateOfBirth(defaultIfBlank(or.getDob(), null));
+        partyBuilder.address(address);
+        partyBuilder.telephoneNumber(telephoneNumber);
```

Comment:
The way we migrate telephone number might not be schema compliant as we always instantiate `TelephoneNumber telephoneNumber` regardless whether telephone is provided or not. That leads to cases like one below:

```
party: {
  telephone: {
    telephoneNumber: null
  }
}
```

To be compliant if user did not provide telephone number we should do the following instead:

```
party: {
  telephone: null
}
```

Reply:
On a second though CCD breaks schema compliance in address fields so maybe not important.